storage/ndb/include/ndbapi/NdbInterpretedCode.hpp

```diff
@@ -195,6 +313,105 @@ class NdbInterpretedCode
+   *
+   * Space required   Buffer    Request message
+   *   add_const_reg  1 word    1 word
+   *   sub_const_reg  1 word    1 word
+   *   lshift_const_reg 1 word  1 word
+   *   rshift_const_reg 1 word  1 word
+   *   mul_const_reg  1 word    1 word
+   *   div_const_reg  1 word    1 word
+   *   and_const_reg  1 word    1 word
+   *   or_const_reg   1 word    1 word
+   *   xor_const_reg  1 word    1 word
+   *   mod_const_reg  1 word    1 word
+   *
+   * @param RegDest Register to store operation result in
+   * @param RegSource1 Register to use as LHS of operator
+   * @param Constant to use as RHS of operator
+   * @return 0 if successful, -1 otherwise
+   */
+  int add_const_reg(Uint32 RegDest, Uint32 RegSource1, Uint16 Constant);
+  int sub_const_reg(Uint32 RegDest, Uint32 RegSource1, Uint16 Constant);
+  int lshift_const_reg(Uint32 RegDest, Uint32 RegSource1, Uint16 Constant);
+  int rshift_const_reg(Uint32 RegDest, Uint32 RegSource1, Uint16 Constant);
+  int mul_const_reg(Uint32 RegDest, Uint32 RegSource1, Uint16 Constant);
+  int div_const_reg(Uint32 RegDest, Uint32 RegSource1, Uint16 Constant);
+  int and_const_reg(Uint32 RegDest, Uint32 RegSource1, Uint16 Constant);
+  int or_const_reg(Uint32 RegDest, Uint32 RegSource1, Uint16 Constant);
+  int xor_const_reg(Uint32 RegDest, Uint32 RegSource1, Uint16 Constant);
+  int mod_const_reg(Uint32 RegDest, Uint32 RegSource1, Uint16 Constant);
+
+  /* Move from heap memory to register
+   * ---------------------------------
+   * These instructions provide possibilities to read the memory
+   * inserted from reads of columns and load it into registers.
+   * One can read 1 byte treated as an Uint8, 2 bytes treated as
+   * Uint16, 4 bytes treated as Uint32 and 8 bytes treated as
+   * Int64.
+   *
+   * *RegDest= *RegSouce1 <operator> *Constant
+   *
+   * Space required   Buffer    Request message
+   *   read_u8_to_reg 1 word    1 word
+   *   read_u16_to_reg 1 word   1 word
+   *   read_u32_to_reg 1 word   1 word
+   *   read_int64_to_reg 1 word 1 word
+   *
+   * @param RegDest Register to store operation result in
+   * @param RegSource1 Register to use as LHS of operator
+   * @param Constant to use as RHS of operator
+   * @return 0 if successful, -1 otherwise
+   */
+  int write_interpreter_output(Uint32 RegValue, Uint32 outputIndex);
+  int convert_size(Uint32 RegSizeDest, Uint32 RegOffset);
+  int write_size_mem(Uint32 RegSize, Uint32 RegOffset);
+  int read_uint8_to_reg_const(Uint32 RegDest, Uint32 memory_offset);
+  int read_uint16_to_reg_const(Uint32 RegDest, Uint32 memory_offset);
+  int read_uint32_to_reg_const(Uint32 RegDest, Uint32 memory_offset);
+  int read_int64_to_reg_const(Uint32 RegDest, Uint32 memory_offset);
+
+  int read_uint8_to_reg_reg(Uint32 RegDest, Uint32 RegOffset);
+  int read_uint16_to_reg_reg(Uint32 RegDest, Uint32 RegOffset);
+  int read_uint32_to_reg_reg(Uint32 RegDest, Uint32 RegOffset);
+  int read_int64_to_reg_reg(Uint32 RegDest, Uint32 RegOffset);
```

Comment:
I noticed that you renamed all read and write operations from u8 to uint8 and so on.
This is a bit inconsistent as the already existant load_const_* are still called load_const_u8 and so on, and that should probably stay as it is as it is documented like that for a while and might break existing users' code.

Also the last one is load_const_u64, while this PR now adds read_int64* and write_int64* instead of unsigned integers.

It is no problem for me as I do not use it at least for now. But I wanted to comment on the inconsistencies.
Max